From: Luigi Baldoni <aloisio@gmx.com>
Date: 2021-08-04 22:14:02 +0200
Subject: Fix compatibility with Qwt 6.2.0
References: https://sourceforge.net/p/linssid/discussion/bugs/thread/3080bf8f4d/
Upstream: submitted

Disambiguate function calls after new overloads added in qwt 6.2.
(courtesy of Jari Kuittinen)

---

Index: MainForm.cpp
===================================================================
--- a/MainForm.cpp
+++ b/MainForm.cpp
@@ -28,6 +28,7 @@
 #include <qwt_plot_curve.h>
 #include <qwt_plot_grid.h>
 #include <qwt_symbol.h>
+#include <qwt_text.h>
 #include <qwt_plot_marker.h>
 #include <QPointF>
 #include <qwt_scale_draw.h>
@@ -1036,7 +1037,7 @@ void MainForm::fillPlots() {
                     MainForm::cellDataRay[tbi].signal ) );
             }
         } else {
-            MainForm::cellDataRay[tbi].pBandCurve->setSamples(0, 0, 0);
+            MainForm::cellDataRay[tbi].pBandCurve->setSamples((double *)nullptr, (double *)nullptr, 0);
         }
         // now the signal history plot
         int ixStart;
@@ -1055,7 +1056,7 @@ void MainForm::fillPlots() {
                     &(MainForm::cellDataRay[tbi].pHistory->signal[ixStart]), ixLength);
         } else {
 
-            MainForm::cellDataRay[tbi].pTimeCurve->setSamples(0, 0, 0);
+            MainForm::cellDataRay[tbi].pTimeCurve->setSamples((double *)nullptr, (double *)nullptr, 0);
         }
     }
     MainForm::mainFormWidget.chan24Plot->replot();
@@ -1424,4 +1425,4 @@ int MainForm::MinIntStr(const string &s)
 //    const std::string whiteSpaces( " \f\n\r\t\v" );
 //    std::string::size_type pos = str.find_first_not_of( whiteSpaces );
 //    str.erase( pos + 1 );    
-//}
\ No newline at end of file
+//}
